web/src/message_viewport.ts

```diff
@@ -75,12 +75,21 @@ export function get_bottom_whitespace_height(): number {
     return height() * 0.4;
 }
 
+export function should_fetch_more_messages_for_view(): boolean {
+    const last_message_scroll_position =
+        $scroll_container.prop("scrollHeight") - get_bottom_whitespace_height();
+    const bottom = scrollTop() + height();
+    // Check if we are 2/3 of the way through the visible messages.
+    return last_message_scroll_position - bottom < last_message_scroll_position / 3;
```

Comment:
How did you select this algorithm? I'm not sure what properties you're trying to achieve.

Reply:
Okay, that does look like a much better approach, I forgot we were tracking `render_window`.

Reply:
Updated